torch/_library/autograd.py

```diff
@@ -102,9 +123,37 @@ def backward(ctx, *grads):
     ):
         Generated = supports_tensorlist(Generated)
 
+
+
+    Generated_jvp = type(
+        name + "_jvp",
+        (torch.autograd.function._SingleLevelFunction,),
```

Comment:
Here's an alternate proposal (that may or may not work, so save your code):
- The user provides us jvp and setup_context_jvp
- We use this to construct a jvp staticmethod in the `Generated` autograd.Function (so no separate Generated_jvp autograd.Function)
- Don't change the logic for `autograd_impl` (well, you might need to [stop it from skipping Generated.apply](https://www.internalfb.com/code/fbsource/[8341003fed135b1632624b6a2860bb3541944d8b]/fbcode/caffe2/torch/_library/autograd.py?lines=108-109))